frontend/apps/pablo/defi/subsquid/swaps/helpers.ts

```diff
@@ -1,8 +1,49 @@
 import { ChartRange, processSubsquidChartData } from "@/defi/utils";
 import BigNumber from "bignumber.js";
+import moment from "moment";
 import { queryPoolTransactionsByType } from "../pools/queries";
 import { query24hOldTransactionByPoolQuoteAsset } from "./queries";
 
+export function getChartLabels(
+  chartSeries: [number, number][],
+  chartRange: ChartRange
+): string[] {
+  let FORMAT_1D = "hh:mm";
+  let FORMAT_1w = "DD/MM";
+  let FORMAT_1M = "MM/YYYY";
+
+  let MAX_LABELS = 5;
+
+  if (chartSeries.length < MAX_LABELS) {
+    return chartSeries.map((i) =>
+      moment(i[0]).format(
+        chartRange === "24h"
+          ? FORMAT_1D
+          : chartRange === "1w"
+          ? FORMAT_1w
+          : FORMAT_1M
```

Comment:
is it possible to avoid nested ternary and replace with a pure function?

```tsx
function getDateFormat(range) {
  return {
    24h: FORMAT_1D,
    1w: FORMAT_1w,
    1m: FORMAT_1M,
  }[range];
}

// and then...
const date_format = getDateFormat(chartRange);
chartSeries.map((i) => 
  moment(i[0]).format(date_format)
  
```